src/test/java/com/amihaiemil/eoyaml/spec/SpecExamplesITCase.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2016-2020, Mihai Emil Andronache
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation
+ *  and/or other materials provided with the distribution.
+ * Neither the name of the copyright holder nor the names of its
+ *  contributors may be used to endorse or promote products derived from
+ *  this software without specific prior written permission.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+package com.amihaiemil.eoyaml.spec;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.amihaiemil.eoyaml.YamlSequence;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * It tests the spec 1.2 examples.
+ *
+ * @author Aislan Nadrowski (aislan.nadrowski@gmail.com)
+ * @version $Id$
+ * @sinve 4.3.6
+ */
+public final class SpecExamplesITCase {
+
+    /**
+     * It tests the example 2.1 of the Yaml Spec 1.2.
+     *
+     * @throws FileNotFoundException If the file was not found.
+     * @throws IOException If something goes wrong.
+     */
+    @Test
+    public void readFirstSpecExample()
+            throws FileNotFoundException, IOException {
```

Comment:
@ANadrowski In the tests you can just say ``throws Exception``, to avoid noise. It doesn't matter what kind of exception is thrown, the test will fail and we'll see it :)

Reply:
Hi @amihaiemil the problem here is that using just "throws Exception" some errors occurs when mvn clean install -Pcheckstyke, please see it:

![error](https://user-images.githubusercontent.com/18332617/81823072-49e24a80-950a-11ea-9cad-120b8a7e73a5.png)

Any idea?

Reply:
@ANadrowski Those error occur because you specified ``@throws Exception`` in the JavaDoc, but the test signature said ``throws FileNotFoundException,...``. 

If you say:

```java
/**
 * Some unit test.
 * @throws Exception If something goes wrong.
 */
public void test() throws Exception {
    //test code here
}
```
it will work fine :)

Reply:
nice :)